chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+"""Icom IC-R6 Driver"""
+# Copyright 2024 John Bradshaw Mi0SYN <john@johnbradshaw.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# TODO:
+# Channel: Frequency setting
+# Channel: TSQL freq or DTCS code with polarity
+# Channel: Skip scan settings
+# Channel: Comment (not sure if supported in ICF)
+# Banks - channel assignment and name editing
+# Validate TS by freq/band (e.g. 9k TS is only for AM brdcast. Airband?)
+#   CS-R6 shows red X if setting invalid, use that to verify Chirp-output ICF
+# Scan edges per manual
+# Radio settings - lamp, delays, keytones, ...
+
+# Notes:
+#
+# USA models have WX alert and block certain frequencies (see manual).
+
+from chirp.drivers import icf
+from chirp import chirp_common, directory, bitwise
+from chirp.settings import RadioSettingGroup, RadioSetting, \
+    RadioSettingValueBoolean, RadioSettingValueList, \
+    RadioSettingValueString, \
+    RadioSettingValueFloat, RadioSettings
+#    RadioSettingValueInteger, \
+
+mem_format = """
+// Channel memories: 1300x 16-byte blocks
+// 0x000 to 0x513F inclusive
+struct {
+  u8 freq0;               // Freq: low byte
+  u8 freq1;               // Freq: mid byte
+  u8 freq_flags:6,        // Related to multiplier (step size)
+     freq2:2;             // Freq: high bits - 18 bits total = step count
```

Comment:
I'd really prefer that you use a u24 here instead of all the bit-shifting. I think you can just mask out the flag bits yourself once you grab the value. I'll do some thinking on how I could make this define-able in the bitwise here...

Reply:
On the to-do list, I need to figure out the endians, radio uses both.